code/modules/projectiles/ammunition.dm

```diff
@@ -62,127 +67,133 @@
-	if(!bullets_from || !istype(bullets_from)) //fuck you for calling this with the wrong arguments
-		return 0
-	if(!target || !istype(target))
-		return 0
-	var/trying_to_load = 0
+		user << "<span class='notice'>You remove \a [dropped.name] from \the [src].</span>"
+
+//We load a target item full from our ammo storage item
+//This loads the bullets one by one, but it's automated so you don't get carpal tunnel
+//Includes a delay and strings
+/obj/item/ammo_storage/proc/load_from(var/obj/item/ammo_storage/bullets_from, var/obj/item/target, var/mob/living/user)
+
+	var/trying_to_load = 0 //We need to figure out how many bullets we want to load, first of all
 	if(istype(target, /obj/item/weapon/gun/projectile))
 		var/obj/item/weapon/gun/projectile/PW = target
-		trying_to_load = min(PW.max_shells - PW.loaded.len, bullets_from.stored_ammo.len) //either we fill to max, or we fill as much as possible
+		trying_to_load = min(PW.max_shells - PW.loaded.len, bullets_from.stored_ammo.len) //Either we fill to max, or we fill as much as possible
 	else
 		var/obj/item/ammo_storage/AS = target
-		trying_to_load = min(AS.max_ammo - AS.stored_ammo.len, bullets_from.stored_ammo.len) //either we fill to max, or we fill as much as possible
-	if(usr && trying_to_load)
-		usr << "You begin loading \the [target]..."
-	if(trying_to_load && do_after(usr,target,trying_to_load * 5)) //bit of a wait, but that's why it's SLOW
-		return 1
-	else if(trying_to_load)
-		var/dropped_bullets = 0
-		var/to_drop = rand(1, trying_to_load) //yeah, drop some on the floor!
-		for(var/i = 1; i<=min(to_drop, bullets_from.stored_ammo.len); i++)
-			var/obj/item/ammo_casing/AC = bullets_from.stored_ammo[1]
-			bullets_from.stored_ammo -= AC
-			AC.loc = get_turf(target)
-			dropped_bullets++
-			bullets_from.update_icon()
-		if(usr)
-			usr << "<span class='rose'>You fumble around and drop [dropped_bullets] shell\s!</span>"
-		return 0
-	return 0
-
-//used to load bullets from ammo storage into other ammo storage or guns
-//bullets_from is the origin, target is the gun or targetted box
-/obj/item/ammo_storage/proc/LoadInto(var/obj/item/ammo_storage/bullets_from, var/obj/item/target)
+		trying_to_load = min(AS.max_ammo - AS.stored_ammo.len, bullets_from.stored_ammo.len) //Either we fill to max, or we fill as much as possible
+	if(trying_to_load) //Non-zero amount
+		user.visible_message("<span class='notice'>[user] begins loading \his [target] from \a [bullets_from.name].", \
+		"<span class='notice'>You begin loading your [target] from \a [bullets_from.name].")
+
+	var/loaded_tally = 0
+	for(var/i = 1; i <= trying_to_load; i++) //For every single round we are going to load
+		if(bullets_from == user.get_active_hand() && get_dist(user, target) <= 1 && get_dist(bullets_from, target) <= 1) //Well, we can't use do_after, so plan B
+			sleep(bullets_from.loading_delay)

```

Comment:
You mean if I throw everything on the floor during this delay I use telepathy magicks?